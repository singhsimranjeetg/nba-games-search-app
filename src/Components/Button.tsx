import styled, { css } from "styled-components";

const buttonStyles = css`
  display: flex;
  align-items: center;
  padding: 0.5rem 1rem;
  border-radius: 100px;
  border: none;
  transition: box-shadow 300ms ease;
  cursor: pointer;
  justify-content: center;
  color: var(--white);
  width: min-content;
  user-select: none;

  &:disabled {
    background-color: var(--surface);
  }

  &:focus {
    outline: none;
  }

  @media (min-width: 769px) {
    &:hover {
      box-shadow: var(--hover);
    }
  }
`;

const PrimaryButton = styled.button`
  ${buttonStyles}
  background-color: var(--green);
`;

const SecondaryButton = styled.button`
  ${buttonStyles}
  background-color: var(--black);
`;

export type ButtonProps = {
  secondary?: boolean;
  disabled?: boolean;
  children?: React.ReactFragment | HTMLCollection | string;
  onClick: (e?: React.MouseEvent) => void;
} & React.ButtonHTMLAttributes<HTMLButtonElement>;

const Button = ({
  secondary,
  disabled,
  onClick,
  children,
  style,
  id,
  ...other
}: ButtonProps) =>
  secondary ? (
    <SecondaryButton
      style={style}
      disabled={disabled}
      onClick={onClick}
      id={id}
      {...other}
    >
      <strong>{children}</strong>
    </SecondaryButton>
  ) : (
    <PrimaryButton
      style={style}
      disabled={disabled}
      onClick={onClick}
      id={id}
      {...other}
    >
      <strong>{children}</strong>
    </PrimaryButton>
  );

export default Button;
