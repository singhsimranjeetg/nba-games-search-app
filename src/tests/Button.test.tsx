import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Components/Button';

it('Snapshot tests Button', () => {
  expect(shallow(<Button onClick={() => {}} />)).toMatchSnapshot();
});
