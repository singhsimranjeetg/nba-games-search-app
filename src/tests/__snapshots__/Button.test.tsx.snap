// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Button 1`] = `
<styled.button
  onClick={[Function]}
>
  <strong />
</styled.button>
`;
